test/unit/app.spec.js

```diff
@@ -37,6 +37,26 @@ describe('App', function() {
     });
   });
 
+  describe('when restarting the app', function() {
```

Comment:
this test would pass with or without calling `this.myApp.restart();`  We need to test that the app stops and starts up again. The state passed to `onBeforeStart` should be the updated state.